packages/common/src/converter/data-converter.ts

```diff
@@ -1,3 +1,4 @@
+import { DefaultFailureConverter, FailureConverter } from '../failure';
```

Comment:
I have a bit of a circular import problem if I do this in Python and it can be a bit confusing even here in TS. To have `converter/data-converter.ts` depend on `failure.ts` which depends on `converter/payload-converters.ts`. This is fine in TS because you are breaking this stuff out into so many files (which I don't necessarily agree with), but it's still confusing to have thing in subdir depend on thing in general dir that depends on thing back in subdir.

I think it might be clearer to not have `failure.ts` depend on anything from `converter` and move anything related to error conversion (i.e. needing error _and_ converter stuff) into things in this directory. But if you don't want to create a `converter/failure-converter.ts`, no worries.

I think in Python at least, I'll have `temporalio.converter.FailureConverter` right next to `temporalio.converter.PayloadConverter` and `temporalio.converter.DataConverter`, and remove all `temporalio.converter` imports from `temporalio.exceptions`. Today `temporalio.exceptions` depends on `temporalio.converter` but I think I'm gonna flip that now that `DataConverter` will accept a `FailureConverter` (class). I haven't written it yet, but it makes the most sense. Thoughts welcome.

Reply:
> But if you don't want to create a converter/failure-converter.ts, no worries.

This is what I did initially and ended up moving the code into `failure.ts`, I can definitely move the converter code to its own file.

Reply:
Fine here if you don't mind the dependency graph it creates.

In Python I can't have `DataConverter` depending on a separate-package `FailureConverter` that depends back on the data-converter package for `PayloadConverter`.